fe/fe-core/src/main/java/org/apache/doris/load/DeleteHandler.java

```diff
@@ -162,14 +173,46 @@ public void process(DeleteStmt stmt) throws DdlException, QueryStateException {
                 }
 
                 if (noPartitionSpecified) {
+                    // Try to get selected partitions if no partition specified in delete statement
+                    // Use PartitionPruner to generate the select partitions
                     if (olapTable.getPartitionInfo().getType() == PartitionType.RANGE
                             || olapTable.getPartitionInfo().getType() == PartitionType.LIST) {
-                        if (!ConnectContext.get().getSessionVariable().isDeleteWithoutPartition()) {
-                            throw new DdlException("This is a range or list partitioned table."
-                                    + " You should specify partition in delete stmt,"
-                                    + " or set delete_without_partition to true");
+                        Set<String> partitionColumnNameSet = Utils.execWithReturnVal(
```

Comment:
Why not just call `olapTable.getPartitionColumnNames()`?
I think we should not eliminate those exceptions.